src/app/execution/txRunner.js

```diff
@@ -142,62 +138,16 @@ TxRunner.prototype.runInNode = function (from, to, data, value, gasLimit, useCal
     if (self._api.config.getUnpersistedProperty('doNotShowTransactionConfirmationAgain')) {
       return executeTx(tx, null, self._api, callback)
     }
+
     self._api.detectNetwork((err, network) => {
       if (err) {
         console.log(err)
         return
       }
-      if (network.name !== 'Main') {
-        return executeTx(tx, null, self._api, callback)
-      }
 
-      var amount = executionContext.web3().fromWei(typeConversion.toInt(tx.value), 'ether')
-      var content = confirmDialog(tx, amount, gasEstimation, self,
-        (gasPrice, cb) => {
-          let txFeeText, priceStatus
-          // TODO: this try catch feels like an anti pattern, can/should be
-          // removed, but for now keeping the original logic
-          try {
-            var fee = executionContext.web3().toBigNumber(tx.gas).mul(executionContext.web3().toBigNumber(executionContext.web3().toWei(gasPrice.toString(10), 'gwei')))
-            txFeeText = ' ' + executionContext.web3().fromWei(fee.toString(10), 'ether') + ' Ether'
-            priceStatus = true
-          } catch (e) {
-            txFeeText = ' Please fix this issue before sending any transaction. ' + e.message
-            priceStatus = false
-          }
-          cb(priceStatus, txFeeText)
-        },
-        (cb) => {
-          executionContext.web3().eth.getGasPrice((error, gasPrice) => {
-            var warnMessage = ' Please fix this issue before sending any transaction. '
-            if (error) {
-              return cb('Unable to retrieve the current network gas price.' + warnMessage + error)
-            }
-            try {
-              var gasPriceValue = executionContext.web3().fromWei(gasPrice.toString(10), 'gwei')
-              cb(null, gasPriceValue)
-            } catch (e) {
-              cb(warnMessage + e.message, null, false)
-            }
-          })
-        }
-      )
-
-      modalDialog('Confirm transaction', content,
-        { label: 'Confirm',
-          fn: () => {
-            self._api.config.setUnpersistedProperty('doNotShowTransactionConfirmationAgain', content.querySelector('input#confirmsetting').checked)
-            if (!content.gasPriceStatus) {
-              return callback('Given gas grice is not correct')
-            }
-            var gasPrice = executionContext.web3().toWei(content.querySelector('#gasprice').value, 'gwei')
-            executeTx(tx, gasPrice, self._api, callback)
-          }}, {
-            label: 'Cancel',
-            fn: () => {
-              return callback('Transaction canceled by user.')
-            }
-          })
+      confirmCb(network, tx, gasEstimation, callback, (gasPrice) => {
```

Comment:
not really sure about that, but seems `confirmCb` is declared as `(network, tx, continueTxExecution, gasEstimation, cancelCb)` in `universal dapp`

Reply:
unfortunately test pass :( that still because we test only against the javascript VM)

Reply:
fixed params in the universal-dapp